testing/kuttl/kuttl-test-generated.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestSuite
+testDirs:
+- testing/kuttl/generated/13
```

Comment:
would we want multiple directories with the generated tests :question:  I was thinking we would just have one set of tests committed. If you wanted to test a different version than the one committed you would generate the files with the make target and a new set of versions/images/etc :thought_balloon: We maybe we could even `gitignore` the files so changes aren't annoying in git?

Reply:
(a) multiple directories -- I don't think so, and I've redone the makefile* so we're adding tests into one `generated` folder, so we don't have to keep changing this file as we add new versions

* Argh, there should've been a makefile added
* 
(b) I definitely can see the appeal of not adding versions_supported * files_needed_to_test number of files each time we add an e2e test. Heck, I'm not sure how you've been using KUTTL, but I've been doing single test runs, so maybe we leave the generated folder completely gitignore'd and the user can generate the tests/versions they want to test? (Meanwhile our CI is running all of them.)

Reply:
I am adding tests in the `generated` folder, but more just to see what they would be like.

Reply:
To point (b) - I like the idea of not having anything committed a lot. You would have to run the generate target every time you checkout a new version to ensure you have the latest changes. I guess that could lead to testing with the wrong generated files if I forget :thinking: 

What are your thoughts on generating the files as part of the `make check-kuttl` target? 
```
.PHONY: check-kuttl
check-kuttl: generate-kuttl
	${PGO_KUBE_CLIENT} ${KUTTL_TEST} --config testing/kuttl/kuttl-test.yaml
```
The workflow could be 1) make changes to the template 2) generate and run the tests. This would ensure that changes are always made and work in the template. You could still run tests separately with the make target `KUTTL_TEST='kuttl test --test <test-name>' make check-kuttl`.

Reply:
> I like the idea of not having anything committed a lot.

> What are your thoughts on generating the files as part of the `make check-kuttl` target?

This would be my preference. The separate `generate-kuttl` target (that `check-kuttl` invokes) also accommodates those that run `kuttl` outside of make.

Reply:
OK, I added the generate to the test command; this way you can just run test and it will generate the files (for version 14 only) OR you can run generate separately if you need to check the generated files or produce test files for another version.

And added a .gitignore to prevent any files from going to /generated

Reply:
(and just to be explicit, you can also run the test make target with the alternate env vars, which should generate and test the files in one go)